### ISSUE TITLE
Passing List<Integer> to a Json property

### ISSUE THREAD

**Issue by deepakmittal888** [vincentrussell/json-data-generator#44]
Hi, is there a way to pass List<Integer> to a json property by creating my custom tag?

Example - 

There is a JSON:
{
"attribute1" : "hello",
"attribute2" : []
}

Now, I want to pass a list of integer to "attribute2" by creating my own tag.

Can you please guide me in doing this? Thanks

**Comment by vincentrussell** [comment]
if you want a a list of integers that just put them in the json :

"attribute2" : [1, 2, 4]

I feel like I must be missing something and you wanted to do something more complicated than that.   Can you provide more information?

**Comment by deepakmittal888** [comment]
Those values are dynamic and will get generate at run time. I can't directly pass them in JSON as those are not static values and may get change. There is a logic in place which will generate that list during execution and I've to put that list values in the field.

I hope I am able to describe it better, else, I'll provide a better example if still it's confusing you.

**Comment by vincentrussell** [comment]
Yes.  Please provide an example because you can use functions with repeats with arrays like this:

 friends: [
      '{{repeat(3)}}',
      {
        id: '{{index("inner")}}',
        name: 'nameValue'{{resetIndex("inner")}}
      }
    ],

**Comment by deepakmittal888** [comment]
My JSON is below:

{
Students:[
{
"id": 01,
"name": "First",
"Subjects": []
},
"id": 02,
"name": "Second",
"Subjects": []
},
{
"id": 03,
"name": "First",
"Subjects": []
},
"id": 04,
"name": "Second",
"Subjects": []
}
]
}

Now for every student, subjects are different.
Example - Student with id - 01 should mapped with subjects as 01, 02, 03 (where these IDs belongs to 3 subject names say for suppose "Physics", "Chemistry", "Math"),
Student with id - 02 should mapped with subjects as 03, 04 (04 is for suppose "Biology") and so on.

Now, what subjects should allot to which students, that I'll decide on run-time based on some conditions.

So, I've created a tag where I've written the code that student wit id 01 should map with 01, 02, 03. For that I've to fill that "Subjects" field with these 3 values.

Now, below is the challenge of using "Repeat" tag:
Suppose I am returning single value a time from my tag (For above example suppose it'll return just "01" for the first time) and push that in "Subjects" array in json. Then in second time when that "Repeat" tag run, we'll not sure if my tag will return "02" or it'll again return "01". To maintain this I'll need to write some extra code (May be a list of tracking what IDs are already being passed). I don't want this patch.

So, instead of using "Repeat" tag and return single value at a time from my tag, can't we just have a tag or something which will allow me to return multiple values at a time and fill it in "Subjects" field at once?

I am extremely sorry but I think I won't be able to explain it much better in written than this. If still this doesn't make sense to you, then I'll try to figure out a workaround here.

**Comment by vincentrussell** [comment]
So the data generator is meant mostly to produce "random" data.   It sounds like you want complicated decision-based logic which isn't supported, which is probably a good thing from a complexity standpoint.

the repeat function isn't just for providing one value it should be able to provide multiple values like:

"Subjects": [
'{{repeat(3)}}',
random("Physics","Chemistry","Math")
]
},

but I'm guessing that isn't what you want.
